react-components/src/components/RevealToolbar/SelectSingle3DModelSelection.tsx

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { useState, type ReactElement } from 'react';
+
+import { Button, Dropdown, Menu, Tooltip as CogsTooltip } from '@cognite/cogs.js';
+import { useTranslation } from '../i18n/I18n';
+import styled from 'styled-components';
+import { useAll3dModels } from '../../hooks/useAll3dModels';
+import { type CogniteClient } from '@cognite/sdk';
+import { ModelsList } from './ModelsList';
+import { useRevisions } from '../../hooks/useRevisions';
+import { type ModelWithRevision } from '../../hooks/types';
+
+type Single3DModelSelectionProps = {
+  sdk: CogniteClient;
+  onModelChange: (model: ModelWithRevision | undefined) => void;
+};
+
+export const Single3DModelSelection = ({
+  sdk,
+  onModelChange
+}: Single3DModelSelectionProps): ReactElement => {
+  const { data: models } = useAll3dModels(sdk, true);
+
+  const { data: modelsWithRevision } = useRevisions(sdk, models);
+
+  const [selectedModel, setSelectedModel] = useState<ModelWithRevision | undefined>();
+
+  const { t } = useTranslation();
+
+  const handleSelectedModelChange = (model: ModelWithRevision | undefined): void => {
```

Comment:
Should be a `useCallback`

Reply:
https://github.com/cognitedata/reveal/pull/4746/commits/612d255baa09001a5a8175e151c71699e6943784